components/Select.tsx

```diff
@@ -1,22 +1,26 @@
 import { JSX } from "preact";
+import clsx from "npm:clsx";
 
 export type SelectProps = {
   name: string;
   options: Array<{ value: string; label: string }>;
+  full?: boolean;
   invalid?: boolean;
   disabled?: boolean;
 } & JSX.HTMLAttributes<HTMLSelectElement>;
 
 export default function Select({
   name,
   options,
+  full = false,
   invalid = false,
   disabled = false,
   ...rest
 }: SelectProps) {
   return (
     <select
-      class="border rounded shadow-block p-2 outline-offset-0 focus:outline-none focus:shadow-focus transition duration-200 text-black"
+      class={"border rounded shadow-block p-2 outline-offset-0 focus:outline-none focus:shadow-focus transition duration-200 text-black " +
+        clsx(full && "w-full")}
```

Comment:
[nit]
the more clsx way of doing this would be

```suggestion
      class={clsx(
        "border rounded shadow-block p-2 outline-offset-0 focus:outline-none focus:shadow-focus transition duration-200 text-black ",
        "w-full": full,  
      )}
```